rust/arrow/src/compute/kernels/boolean.rs

```diff
@@ -149,6 +150,64 @@ pub fn is_not_null(input: &ArrayRef) -> Result<BooleanArray> {
     Ok(BooleanArray::from(Arc::new(data)))
 }
 
+/// Copies original array, setting null bit to true if a secondary comparison boolean array is set to true.
+/// Typically used to implement NULLIF.
+pub fn nullif<T>(
+    left: &PrimitiveArray<T>,
```

Comment:
If I understood correctly, because we only use `left.data()`, if we replace `PrimitiveArray<T>` by `impl Array` and remove the generic `T`, this is generalized for all data types. :)

Note that this will not address this in DataFusion, as it seems that we can only compute comparison boolean arrays from primitives atm.

Reply:
Hmmm, I'm trying this, but the return type becomes more complicated....    and as you pointed out, it doesn't help in the end. :-p

Reply:
I think that it would help because on the `arrow` crate this would be a complete implementation. There are people that use `arrow` (but not DataFusion), that would still benefit.

The following would work (I tested it locally :)), if you have the energy:

```rust
pub fn nullif(
    left: &impl Array,
    right: &BooleanArray,
) -> Result<ArrayRef>
...

    // Construct new array with same values but modified null bitmap
    let data = ArrayData::new(
        left.data_type().clone(),
        left.len(),
        None, // Note how I am using `None` here, to make `new` compute the number.
        modified_null_buffer,
        left.offset(),
        left_data.buffers().to_vec(),
        left_data.child_data().to_vec(),
    );
    Ok(make_array(Arc::new(data)))
```

Reply:
@jorgecarleitao see my comment about why I can't make this more generic for now....  so I'm marking this as resolved.  However everything else should be addressed.